### DISCUSSION TITLE
Density profiles

### DISCUSSION THREAD

**Discussion by chraibi** [PedestrianDynamics/PedPy#225] [Ideas]
Density profiles with Voronoi are great and precise. 

However, sometimes you may have a big geometry or a lot of people moving around.
This means the profile calculations can be very slow. 

At that scale, it might be enough to have an imprecise density profile, but fast. 

Therefore, I suggest introducing new density profiles: 

1. [Classical method](https://github.com/PedestrianDynamics/dashboard/blob/main/Utilities.py#L808)
2. [Gaussian method](https://github.com/PedestrianDynamics/dashboard/blob/main/Utilities.py#L936)
    (published in this [paper](https://royalsocietypublishing.org/doi/10.1098/rsos.211822))

You can try both methods [here](go.fzj.de/dashboard).

Here some equations:
<img width="924" alt="Screenshot 2023-09-27 at 21 47 26" src="https://github.com/PedestrianDynamics/PedPy/assets/5772973/c0ff5d85-9f15-4e7a-a81d-9a967429880d">

**Comment by schroedtert** [comment]
The suggested functions to compute the density profiles have been added to _PedPy_. With the next release (1.1) they will available via PyPi, until then you can check them by installing the latest (unstable) version of _PedPy_ via `pip install git+https://github.com/PedestrianDynamics/PedPy.git`.

To use them you can do the following:
```python
from pedpy import compute_density_profile, DensityMethod

grid_size = 0.25

classic_density_profile = compute_density_profile(
    data=traj.data,
    walkable_area=walkable_area,
    grid_size=grid_size,
    density_method=DensityMethod.CLASSIC,
)

gaussian_density_profile = compute_density_profile(
    data=traj.data,
    walkable_area=walkable_area,
    grid_size=grid_size,
    density_method=DensityMethod.GAUSSIAN,
    gaussian_width=0.5,  
)
```

![image](https://github.com/PedestrianDynamics/PedPy/assets/15690524/91b8c6c1-edb3-4747-9b28-6b680b98895e)

Thank you, for the valuable input!
